### PR TITLE
feat: add OCI plugin distribution docs

### PR DESCRIPTION
**Pull request by johnharris85**
### Description

Added docs around using an OCI registry and OSS tooling to sign and distribute custom plugins.

### Testing instructions

Preview link: https://deploy-preview-7245--kongdocs.netlify.app/gateway/latest/plugin-development/distribution/#via-oci-artefacts

### Checklist 

- [X] Review label added 
- [X] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable. **N/A**

### CODE REVIEW

**Comment by mheap** [comment]
The flow matches what I expected. I'd like @rspurgeon to review and see how it fits in to the new custom plugin docs

**Comment by rspurgeon** [comment]
Can we add a link directly to the preview build of the page?  I can't find the updated doc in unreleased or 3.6

**Comment by johnharris85** [comment]
> Can we add a link directly to the preview build of the page? I can't find the updated doc in unreleased or 3.6

Updated

**Comment by sentinelleader** [comment]
> The flow matches what I expected. I'd like @rspurgeon to review and see how it fits in to the new custom plugin docs

@mheap @johnharris85 im a little bit curious about this flow. If im a user with say 4-5 custom plugin

1. i need to create separate ci pipelines per plugin
2. need to find a way to get this deployed and unpacked (if its a k8s based, init container, and if its a vm then based on the deployment system or some other way)

where as if we go with the bundling of image, i can just live with a single CI/CD pipeline and can still control versioning via image tag. Apart from artifact storage and signing, there is still lot of steps involved from a consumer pov. Are we eventually opening door to options like manageable plugin registry or some ?

File: app/_src/gateway/plugin-development/distribution.md [hunk 1]
```diff
@@ -323,6 +328,94 @@ build type to define modules in Lua notation and their corresponding file.
...
+2. Use the OSS [Cosign tool][cosign-install] to generate a key pair for use signing and verifying plugins:
+
+    ```
+    cosign generate-key-pair
+    ```
+
+    The private key (`cosign.key`) should be kept secure and is used for signing the plugin artefact. The public key 
+    (`cosign.pub`) should be distributed and used by target machines to validate the downloaded plugin later in the flow. 
+
+    There are also keyless methods for signing and verifying artefacts with Cosign. More information
```
Comment (Guaris) [inline app/_src/gateway/plugin-development/distribution.md:361]:
```suggestion
    There are also key-less methods for signing and verifying artifacts with Cosign. More information 
```

File: app/_src/gateway/plugin-development/distribution.md [hunk 2]
```diff
@@ -323,6 +328,94 @@ build type to define modules in Lua notation and their corresponding file.
...
+    cosign generate-key-pair
+    ```
+
+    The private key (`cosign.key`) should be kept secure and is used for signing the plugin artefact. The public key 
+    (`cosign.pub`) should be distributed and used by target machines to validate the downloaded plugin later in the flow. 
+
+    There are also keyless methods for signing and verifying artefacts with Cosign. More information 
+    is available in their [documentation][cosign-signing].
+
+3. Login to your OCI-compliant registry. In this case we'll use Dockerhub:
```
Comment (Guaris) [inline app/_src/gateway/plugin-development/distribution.md:364]:
```suggestion
3. Login to your OCI-compliant registry. In this case we'll use Docker Hub:
```

File: app/_src/gateway/plugin-development/distribution.md [hunk 3]
```diff
@@ -323,6 +328,94 @@ build type to define modules in Lua notation and their corresponding file.
...
+
+    ```
+    cat ~/foo_password.txt | docker login --username foo-user --password-stdin
+    ```
+
+4. Upload the plugin artefact to the OCI registry using Cosign. This is the equivalent of running
```
Comment (Guaris) [inline app/_src/gateway/plugin-development/distribution.md:370]:
```suggestion
4. Upload the plugin artifact to the OCI registry using Cosign. This is the equivalent of running 
```

File: app/_src/gateway/plugin-development/distribution.md [hunk 4]
```diff
@@ -323,6 +328,94 @@ build type to define modules in Lua notation and their corresponding file.
...
+    is available in their [documentation][cosign-signing].
+
+3. Login to your OCI-compliant registry. In this case we'll use Dockerhub:
+
+    ```
+    cat ~/foo_password.txt | docker login --username foo-user --password-stdin
+    ```
+
+4. Upload the plugin artefact to the OCI registry using Cosign. This is the equivalent of running 
+a `docker push <image>` when pushing a local Docker image up to a registry.
```
Comment (Guaris) [inline app/_src/gateway/plugin-development/distribution.md:371]:
```suggestion
`docker push <image>` when pushing a local Docker image up to a registry.
```

**Comment by Guaris** [comment]
@rspurgeon have you had time to review this?

File: app/_data/docs_nav_gateway_3.6.x.yml [hunk 5]
```diff
@@ -447,7 +447,7 @@ items:
         url: /plugin-development/admin-api
       - text: Writing Tests
         url: /plugin-development/tests
-      - text: (un)Installing your Plugin
+      - text: Managing your custom plugin
```
Comment (rspurgeon) [inline app/_data/docs_nav_gateway_3.6.x.yml:450]: IMO "Managing" is too generic.  I would suggest "Deploying" as that seems relevant to the general topic here.

File: app/_src/gateway/plugin-development/distribution.md [hunk 6]
```diff
@@ -1,5 +1,5 @@
 ---
-title: (un)Installing your plugin
+title: Managing your custom plugin
```
Comment (rspurgeon) [inline app/_src/gateway/plugin-development/distribution.md:2]: Ditto on my comment in the TOC. IMO "Managing" is too generic.

**Comment by rspurgeon** [comment]
> @rspurgeon have you had time to review this?

@Guaris I made one comment about the title of the document only. The Plugin dev guide quickstart references this page, i'm not sure if this will result in a new URL for this page and we need to update that PR as well. 

https://github.com/Kong/docs.konghq.com/pull/7068

File: app/_data/docs_nav_gateway_3.6.x.yml (hunk 5 above)
Comment (Guaris) [inline app/_data/docs_nav_gateway_3.6.x.yml:450]:
```suggestion
      - text: Installation and Distribution
```

File: app/_data/docs_nav_gateway_3.7.x.yml [hunk 7]
```diff
@@ -443,7 +443,7 @@ items:
         url: /plugin-development/admin-api
       - text: Writing Tests
         url: /plugin-development/tests
-      - text: (un)Installing your Plugin
+      - text: Managing your custom plugin
```
Comment (Guaris) [inline app/_data/docs_nav_gateway_3.7.x.yml:446]:
```suggestion
      - text: Installation and Distribution
```

File: app/_src/gateway/plugin-development/distribution.md (hunk 6 above)
Comment (Guaris) [inline app/_src/gateway/plugin-development/distribution.md:2]:
```suggestion
title: Installation and Distribution
```

**Review by mheap — APPROVED** [review]
Approving to get this released. The whole distribution page needs a rework, which I'll take on

**Pull request merged**